File: openai.py
import openai
from telegram import Update
from telegram.ext import Updater, CommandHandler, MessageHandler, Filters, CallbackContext

# Azure OpenAI settings
OPENAI_API_KEY = "YOUR_AZURE_OPENAI_API_KEY"
OPENAI_ENDPOINT = "https://YOUR_AZURE_OPENAI_ENDPOINT/openai/deployments/YOUR_DEPLOYMENT_NAME/completions?api-version=2023-03-15-preview"
HEADERS = {"Authorization": f"Bearer {OPENAI_API_KEY}", "Content-Type": "application/json"}

# Telegram Bot Token
TELEGRAM_TOKEN = "YOUR_TELEGRAM_BOT_TOKEN"

def chat_with_openai(prompt):
    payload = {
        "prompt": prompt,
        "max_tokens": 150,
        "temperature": 0.7,
    }
    response = requests.post(OPENAI_ENDPOINT, headers=HEADERS, json=payload)
    if response.status_code == 200:
        return response.json().get("choices")[0].get("text").strip()
    else:
        return "Sorry, there was an issue connecting to OpenAI."

def start(update: Update, context: CallbackContext) -> None:
    update.message.reply_text("Hello! I'm your AI bot. Ask me anything!")

def handle_message(update: Update, context: CallbackContext) -> None:
    user_message = update.message.text
    ai_response = chat_with_openai(user_message)
    update.message.reply_text(ai_response)

def main():
    updater = Updater(TELEGRAM_TOKEN)
    dispatcher = updater.dispatcher

    dispatcher.add_handler(CommandHandler("start", start))
    dispatcher.add_handler(MessageHandler(Filters.text & ~Filters.command, handle_message))

    updater.start_polling()
    updater.idle()

if __name__ == "__main__":
    main()

File: openai.js
const TelegramBot = require('node-telegram-bot-api');
const axios = require('axios');

// Azure OpenAI settings
const OPENAI_API_KEY = 'YOUR_AZURE_OPENAI_API_KEY';
const OPENAI_ENDPOINT = 'https://YOUR_AZURE_OPENAI_ENDPOINT/openai/deployments/YOUR_DEPLOYMENT_NAME/completions?api-version=2023-03-15-preview';

// Telegram Bot Token
const TELEGRAM_TOKEN = 'YOUR_TELEGRAM_BOT_TOKEN';

// Initialize the Telegram bot
const bot = new TelegramBot(TELEGRAM_TOKEN, { polling: true });

// Function to chat with Azure OpenAI
async function chatWithOpenAI(prompt) {
    try {
        const response = await axios.post(
            OPENAI_ENDPOINT,
            {
                prompt: prompt,
                max_tokens: 150,
                temperature: 0.7,
            },
            {
                headers: {
                    'Authorization': `Bearer ${OPENAI_API_KEY}`,
                    'Content-Type': 'application/json',
                },
            }
        );
        return response.data.choices[0].text.trim();
    } catch (error) {
        console.error('Error with OpenAI:', error.message);
        return "Sorry, I couldn't connect to OpenAI.";
    }
}

// Handle start command
bot.onText(/\/start/, (msg) => {
    bot.sendMessage(msg.chat.id, "Hello! I'm your AI bot. Ask me anything!");
});

// Handle user messages
bot.on('message', async (msg) => {
    if (msg.text.startsWith('/')) return; // Ignore commands
    const response = await chatWithOpenAI(msg.text);
    bot.sendMessage(msg.chat.id, response);
});
